helm_drydock/drivers/node/maasdriver/driver.py

```diff
@@ -0,0 +1,306 @@
+                            status=hd_fields.TaskStatus.Running)
+
+        site_design = self.orchestrator.get_effective_site(design_id, task.site_name)
+
+        if task.action == hd_fields.OrchestratorAction.CreateNetworkTemplate:
+            subtask = self.orchestrator.create_task(task_model.DriverTask,
+                        parent_task_id=task.get_id(), design_id=design_id,
+                        action=task.action, site_name=task.site_name,
+                        task_scope={'site': task.site_name})
+            runner = MaasTaskRunner(state_manager=self.state_manager,
+                        orchestrator=self.orchestrator,
+                        task_id=subtask.get_id(),config=self.config)
+            runner.start()
+
+            runner.join(timeout=120)
+
+            if runner.is_alive():
+                result =  {
+                    'retry': False,
+                    'detail': 'MaaS Network creation timed-out'
+                }
+                self.orchestrator.task_field_update(task.get_id(),
+                            status=hd_fields.TaskStatus.Complete,
+                            result=hd_fields.ActionResult.Failure,
+                            result_detail=result)
+            else:
+                subtask = self.state_manager.get_task(subtask.get_id())
+                self.orchestrator.task_field_update(task.get_id(),
+                            status=hd_fields.TaskStatus.Complete,
+                            result=subtask.get_result())
+
+            return
+
+class MaasTaskRunner(drivers.DriverTaskRunner):
+
+    def __init__(self, config=None, **kwargs):
+        super(MaasTaskRunner, self).__init__(**kwargs)
+
+        self.driver_config = config
+
+    def execute_task(self):
+        task_action = self.task.action
+
+        self.orchestrator.task_field_update(self.task.get_id(),
+                            status=hd_fields.TaskStatus.Running,
+                            result=hd_fields.ActionResult.Incomplete)
+
+        self.maas_client = MaasRequestFactory(self.driver_config['api_url'],
+                                              self.driver_config['api_key'])
+
+        site_design = self.orchestrator.get_effective_site(self.task.design_id,
+                                                           self.task.site_name)
+
+        if task_action == hd_fields.OrchestratorAction.CreateNetworkTemplate:
+            # Try to true up MaaS definitions of fabrics/vlans/subnets
+            # with the networks defined in Drydock
+            design_networks = site_design.networks
+
+            subnets = maas_subnet.Subnets(self.maas_client)
+            subnets.refresh()
+
+            result_detail = {
+                'detail': []
+            }
+
+            for n in design_networks:
+                exists = subnets.query({'cidr': n.cidr})
+
+                subnet = None
+
+                if len(exists) > 0:
+                    subnet = exists[0]
```

Comment:
Things like this scare me.  I do not know if this is choosing the first element because several can be returned matching the query above or if the element you wish to work with is within the first element in the array returned and the query would never return several subnets even if several existed with the same cidr. If somehow we duplicated subnets (human mistake, code idempotency issue), we would only ever work with the first returned? Should we not expect only a single subnet returned ever